src/Kdyby/Validator/ConstraintValidatorFactory.php

```diff
@@ -47,23 +49,16 @@ public function __construct(Nette\DI\Container $serviceLocator)
 	 */
 	public function getInstance(Constraint $constraint)
 	{
-		$className = $constraint->validatedBy();
+		$name = $constraint->validatedBy();
 
-		// Workaround for https://github.com/symfony/symfony/pull/16166.
-		if ($className === 'validator.expression') {
-			$className = 'Symfony\Component\Validator\Constraints\ExpressionValidator';
+		if (!isset($this->validators[$name])) {
+			trigger_error(sprintf('Please register all you constraint validators as DI services, namely %s.', $name), E_USER_DEPRECATED);
+			$this->validators[$name] = $this->serviceLocator->createInstance($name);
```

Comment:
More importantly with this you might want to reconsider adding all the default validators into DIC. Almost all of them would work just fine with https://github.com/symfony/symfony/blob/6f48ec5be698f541457c6ceb7a84fca60a7c80ad/src/Symfony/Bundle/FrameworkBundle/Validator/ConstraintValidatorFactory.php#L71.